### PR TITLE
Tests for removing special characters

### PR DESCRIPTION
**Pull request by AnnaLangner**
Tests for removing special characters from phone numbers

### CODE REVIEW

**Comment by AnnaLangner** [comment]
The tests pass but I get the information: `unable to open database file`
What does it mean?

**Review by klangner — COMMENTED** [review]
You need this code in the main.py
```python
if __name__ == "__main__":
    main()
```
To understand why, check:
https://www.guru99.com/learn-python-main-function-with-examples-understand-main.html

File: script/main_test.py [hunk 1]
```diff
@@ -1,25 +1,57 @@
...
-# class TestRemovesSpecialCharacters(unittest.TestCase):
+class TestRemovesSpecialCharacters(unittest.TestCase):
 
-#   def test_remove_special_characters(self):
-#     dict1 = {
-#       "phone":"04-56-18-88-34",
-#       "cell":"06-74-93-14-75"
-#     }
-#     list_dict = [dict1]
+  def test_remove_special_characters(self):
```
Comment (klangner) [inline script/main_test.py:7]: This function should looks like this:
```python
def test_remove_special_characters(self):
    self.assertEqual(main.clean_phone_number("04-56-18-88-34"), "0456188834")
```

But the problem here is that there is no such function `clean_phone_number`. Instead you have only function `remove_special_characters_from_phone_numbers` which is hard to test.
Refactor the code in the main.py. Create this new function and use it inside `remove_special_characters_from_phone_numbers` to clean phone numbers.

File: script/main_test.py [hunk 2]
```diff
@@ -1,25 +1,57 @@
 import unittest
+import main
```
Comment (klangner) [inline script/main_test.py:2]: Check https://www.guru99.com/learn-python-main-function-with-examples-understand-main.html to understand why you have this error now.
The solution is:
```python
if __name__ == "__main__":
    main()
```

**Review by klangner — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -46,8 +46,12 @@ This project help you in easy way to pull up information about users in your dat
 
 *If You want to run tests:*
 
+* You must be in your scripts folder,
+
 * **python -m unittest -v main_test.TestDigitAddition**, Run a sample test
```
Comment (klangner) [inline README.md]: You don't need separate lines to run different test.
Find the way to run all the tests at once.

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/main.py [hunk 4]
```diff
@@ -59,16 +59,18 @@ def add_field_time_until_birthday(dob_fields):
...
   for field in phone_fields:
-    phone = field['phone']       
-    clear_phone = re.sub(r'\(|\)|\-|\+|\s', '', phone)
-    field['phone'] = clear_phone
-
-    cell = field['cell'] 
-    clear_cell = re.sub(r'\(|\)|\-|\+|\s', '', cell)
-    field['cell'] = clear_cell
+    phone = field['phone']        
+    field['phone'] = clear_phone_numbers(phone)
```
Comment (klangner) [inline script/main.py]: Better:
```python
field['phone'] = clear_phone_numbers(field['phone'])
```

File: script/main_test.py [hunk 5]
```diff
@@ -1,32 +1,73 @@
...
-#     self.assertEquals("0674931475", list_dict[0]["cell"])   
 
-class TestDigitAddition(unittest.TestCase):
-  def test_addition(self):
-    number1 = 2
-    number2 = 4
-    sum_numbers = number1 + number2
-    self.assertTrue(6, sum_numbers)
+  def test_are_no_special_characters(self):
+    dict1 = {
```
Comment (klangner) [inline script/main_test.py]: What are those dicts for?
Do you need them?

File: script/main_test.py [hunk 6]
```diff
@@ -1,32 +1,73 @@
 import unittest
+import main
 
 
-structure_to_test =[{"phone":"04-56-18-88-34","cell":"06-74-93-14-75"},{"phone":"17533450","cell":"67583818"},{"phone":"075 831 68 55","cell":"075 292 22 27"},{"phone":"(394)-381-6746","cell":"(361)-356-8993"},{"phone":"(00) 1170-0136","cell":"(32) 7367-6536"}]
+class TestRemovesSpecialCharacters(unittest.TestCase):
 
+  def test_remove_special_characters(self):
+    dict1 = {
```
Comment (klangner) [inline script/main_test.py]: What are those dicts for?
Do you need them?

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/main_test.py (hunk 5 above)
Comment (AnnaLangner) [reply script/main_test.py]: Now I dont :)

File: script/main_test.py [hunk 7]
```diff
@@ -1,32 +1,37 @@
...
 
-class TestDigitAddition(unittest.TestCase):
-  def test_addition(self):
-    number1 = 2
-    number2 = 4
-    sum_numbers = number1 + number2
-    self.assertTrue(6, sum_numbers)
+  def test_are_no_special_characters(self):
+
+    self.assertEqual(main.clear_phone_numbers("0456188834"),"0456188834")
```
Comment (klangner) [inline script/main_test.py:24]: Isn't the input and expected data the same here? 
Both are: "0456188834"
What does it test?

File: script/main.py [hunk 8]
```diff
@@ -59,16 +59,16 @@ def add_field_time_until_birthday(dob_fields):
     field["dob"].update(dob_new_field_time_until_birthday)
   
 
-def remove_special_characters_from_phone_numbers(phone_fields):
-  for field in phone_fields:
-    phone = field['phone']       
-    clear_phone = re.sub(r'\(|\)|\-|\+|\s', '', phone)
-    field['phone'] = clear_phone
+def clear_phone_numbers(phone):
```
Comment (klangner) [inline script/main.py]: Since it clears only 1 number, it would be better to name this function as singular `clear_phone_number`

**Review by klangner — CHANGES_REQUESTED** [review]

File: script/main_test.py (hunk 7 above)
Comment (AnnaLangner) [reply script/main_test.py:24]: there are phone numbers in the file that do not contain special characters, I thought to also test them if the function does not destroy something

Comment (klangner) [reply script/main_test.py:24]: This is a very good idea. But 1-2 asserts of this type would be enough

**Review by klangner — CHANGES_REQUESTED** [review]

**Review by klangner — APPROVED** [review]

**Pull request merged**